group_by.py

```diff
@@ -10,14 +10,13 @@ def group_by(function: Optional[callable], iterable: Iterable) -> dict:
     :param iterable: Items to run the function on.
     :return: Dictionary base on the results of the function and the iterable.
     """
-    # return_dict = {}
-    # for item in iterable:
-    #     if function(item) in return_dict.keys():
-    #         return_dict[function(item)].append(item)
-    #     else:
-    #         return_dict[function(item)] = item
-    return {function(item): item for item in iterable}
-    # return return_dict
+    return_dict = {}
+    for item in iterable:
+        if function(item) in return_dict.keys():
+            return_dict[function(item)].extend([item])
+        else:
+            return_dict[function(item)] = [item]
```

Comment:
I would recommend using default_dict - an elegant and efficient way of inserting unknown keys into a dictionary.
https://realpython.com/python-defaultdict/#:~:text=However%2C%20the%20defaultdict%20version%20is,instead%20of%20a%20standard%20dict%20.